packages/router/src/grpc/view-protocol-server/notes.test.ts

```diff
@@ -15,39 +14,58 @@ import {
   NotesResponse,
   SpendableNoteRecord,
 } from '@buf/penumbra-zone_penumbra.bufbuild_es/penumbra/view/v1/view_pb';
+import {IndexedDbMock, MockServices} from "./test-utils";
+import {Services} from "@penumbra-zone/services";
 
 describe('Notes request handler', () => {
-  let mockServices: Services;
+  let mockServices: MockServices;
   let mockCtx: HandlerContext;
+  let mockIndexedDb: IndexedDbMock;
+  let notesNext: Mock;
 
   beforeEach(() => {
     vi.resetAllMocks();
 
+    notesNext = vi.fn();
+    const mockTransactionInfoSubscription = {
+      next: notesNext,
+      [Symbol.asyncIterator]: () => mockTransactionInfoSubscription,
+    };
+
+    mockIndexedDb = {
+      iterateSpendableNotes: () => mockTransactionInfoSubscription,
+    };
+
     mockServices = {
-      getWalletServices: () =>
-        Promise.resolve({
-          indexedDb: {
-            getAllSpendableNotes: (): Promise<SpendableNoteRecord[]> => Promise.resolve(testData),
-          },
-        }),
-    } as Services;
+      getWalletServices: vi.fn(() => Promise.resolve({ indexedDb: mockIndexedDb })),
+    };
 
     mockCtx = createHandlerContext({
       service: ViewService,
-      method: ViewService.methods.balances,
+      method: ViewService.methods.notes,
       protocolName: 'mock',
       requestMethod: 'MOCK',
-      contextValues: createContextValues().set(servicesCtx, mockServices),
+      contextValues: createContextValues().set(servicesCtx, mockServices as unknown as Services),
+    });
+
+    for (const record of testData) {
+      notesNext.mockResolvedValueOnce({
+        value: { value: record },
+      });
+    }
+    notesNext.mockResolvedValueOnce({
+      done: true,
     });
   });
 
   test('empty request return no spend notes', async () => {
+
     const responses: NotesResponse[] = [];
     const req = new NotesRequest({});
     for await (const res of notes(req, mockCtx)) {
       responses.push(new NotesResponse(res));
     }
-    expect(responses.length).toBe(5);
+    expect(responses.length).toBe(6);
```

Comment:
records in testData array have always been 6, but this test was passed before, how?